src/handlers.ts

```diff
@@ -101,23 +113,91 @@ export async function handleChromiumCheck(): Promise<void> {
+        path: 'DEPS',
+        ref: 'master',
+      });
+      const deps = Buffer.from(depsData.data.content, 'base64').toString('utf8');
+      const hashRegex = new RegExp(`${ROLL_TARGETS.CHROMIUM.key}':\n +'(.+?)',`, 'm');
+      const [, chromiumHash] = hashRegex.exec(deps);
+      const lkgr = await getChromiumLkgr();
+      if (chromiumHash !== lkgr.commit) {
+        d(`updating master from ${chromiumHash} to ${lkgr.commit}`);
+        try {
+          await roll({
+            rollTarget: ROLL_TARGETS.CHROMIUM,
+            electronBranch: masterBranch,
+            targetVersion: lkgr.commit,
+          });
+        } catch (e) {
+          d(`Error rolling ${masterBranch.name} to ${lkgr.commit}`, e);
+          thisIsFine = false;
+        }
       }
+    } else {
+      d('master branch not found!');
+    }
+  }
+
+  if (!thisIsFine) {
+    throw new Error(`One or more upgrade checks failed; see the logs for details`);
+  }
+}
+
+export async function handleNodeCheck(): Promise<void> {
+  let thisIsFine = true;
+  const d = debug('roller/node:handleNodeCheck()');
+  const github = await getOctokit();
+
+  d('fetching nodejs/node releases');
+  const { data: releases } = await github.repos.listReleases({
+    owner: REPOS.NODE.OWNER,
+    repo: REPOS.NODE.NAME,
+  });
+  const releaseTags = releases.map((r) => r.tag_name);
+
+  d('fetching electron/electron branches');
+  const { data: branches } = await github.repos.listBranches({
+    owner: REPOS.ELECTRON.OWNER,
+    repo: REPOS.ELECTRON.NAME,
+    protected: true,
+  });
+
+  // TODO: Implement node roller rules for release branches
+  const masterBranch = branches.find((branch) => branch.name === 'master');
+
+  d(`getting DEPS for branch ${masterBranch} in electron/electron`);
+  const depsData = await github.repos.getContents({
+    owner:  REPOS.ELECTRON.OWNER,
+    repo: REPOS.ELECTRON.NAME,
+    path: 'DEPS',
+    ref: 'master',
+  });
+  const deps = Buffer.from(depsData.data.content, 'base64').toString('utf8');
+
+  // find node version from DEPS
+  const versionRegex = new RegExp(`${ROLL_TARGETS.NODE.key}':\n +'(.+?)',`, 'm');
+  const [, depsNodeVersion] = versionRegex.exec(deps);
+  const majorVersion = semver.major(semver.clean(depsNodeVersion));
+
+  d(`computing latest upstream version for Node ${majorVersion}`);
+  const latestUpstreamVersion = semver.maxSatisfying(releaseTags, `^${majorVersion}`);
+
+  // only roll for LTS release lines of Node.js (even-numbered major versions)
+  if (majorVersion % 2 === 0 && semver.gt(latestUpstreamVersion, depsNodeVersion)) {
+    d(`branch ${masterBranch.name} could upgrade from ${depsNodeVersion} to ${latestUpstreamVersion}`);
+    try {
+      await roll({
+        rollTarget: ROLL_TARGETS.NODE,
+        electronBranch: masterBranch,
+        targetVersion: latestUpstreamVersion,
+      });
+    } catch (e) {
+      d(`Error rolling ${masterBranch.name} to ${latestUpstreamVersion}`, e);
+      thisIsFine = false;
```

Comment:
the pattern with `thisIsFine` isn't relevant in this function, as there's no loop. The reason for it in the above function was to prevent a failure in one roll from affecting the others, but here there's only one roll so no reason not to just throw directly.